eureka2-client/src/main/java/com/netflix/eureka2/client/resolver/EurekaServerResolver.java

```diff
@@ -119,20 +131,34 @@ public EurekaServerResolver(
      * exactly one item with list of retrieved servers, or an empty list, or an error.
      */
     private Observable<Set<Server>> fetchFreshServerSet() {
-        return snapshotInterestChannel
-                .forSnapshot(readServerInterest)
-                .reduce(new HashSet<Server>(), new Func2<Set<Server>, InstanceInfo, Set<Server>>() {
-                    @Override
-                    public Set<Server> call(Set<Server> servers, InstanceInfo newInstanceInfo) {
-                        InetSocketAddress socketAddress = serviceSelector.returnServiceAddress(newInstanceInfo);
-                        Server newServer = new Server(socketAddress.getHostString(), socketAddress.getPort());
-                        servers.add(newServer);
-                        return servers;
-                    }
-                })
-                .retryWhen(new RetryStrategyFunc(500, 2, true))
-                .concatWith(Observable.just(Collections.<Server>emptySet()))
-                .take(1);
+        final EurekaClient eurekaClient = eurekaClientBuilder.build();
+        try {
```

Comment:
Try/catch is a safeguard here if the observable creation fails for whatever reason. In such case the doOnTerminate will never be executed, as this observable do not exist yet nor is subscribed to.
It is to guarantee that we release resources (shutdown eureka client).